src/core/device_models.jl

```diff
@@ -0,0 +1,78 @@
+"""
+Abstract type for Device Formulations (a.k.a Models)
+
+# Example
+```julia
+import PowerSimulations
+const PSI = PowerSimulations
+struct MyCustomFormulation <: PSI.AbstractDeviceFormulation
+```
+"""
+abstract type AbstractDeviceFormulation end
+
+"""Formulation that fixes the injection values of devices"""
+struct FixedOutput <: AbstractDeviceFormulation end
+
+function _check_device_formulation(
+    ::Type{D},
+) where {D <: Union{AbstractDeviceFormulation, PSY.Device}}
+    if !isconcretetype(D)
+        throw(
+            ArgumentError(
+                "The device model must contain only concrete types, $(D) is an Abstract Type",
+            ),
+        )
+    end
+end
+
+"""
+Establishes the model for a particular device specified by type. Uses the keyword argument
+feedforward to enable passing values between operation model at simulation time
+
+# Arguments
+-`::Type{D}`: Power System Device Type
+-`::Type{B}`: Abstract Device Formulation
+
+# Accepted Key Words
+- `feedforward::Array{<:AbstractAffectFeedForward}` : use to pass parameters between models
+
+# Example
+```julia
+thermal_gens = DeviceModel(ThermalStandard, ThermalBasicUnitCommitment),
+```
+"""
+mutable struct DeviceModel{D <: PSY.Device, B <: AbstractDeviceFormulation}
```

Comment:
The more I use large systems, the more I think we need some additional control on these definitions. We should consider adding a `filter` field that can be applied in `get_components(component_type, sys, filter)`. Of course, we'd also need a check to issue warnings if you inadvertently fail to cover all components.

Reply:
There is now a possibility to do similar things for the device model. Implementing that filter in this new code won't be too hard but it will effectively NOT model those devices.